### PR TITLE
Updates for release packages

### PR DESCRIPTION
**Pull request by lucianomartin**
This PR contains the following changes:
  - add more information in top-level readme file
  - add top-level readme and license files to firmware release package, see in the [GitHub action](https://github.com/xmos/sln_voice/actions/runs/7060202271). 
  - reworking of the doc packages: the one for the release is in the Jenkins artifacts: [xcore_voice_docs_release.zip](http://srv-bri-jcim0:8080/job/XMOS/job/sln_voice/job/PR-344/2/artifact/xcore_voice_docs_release.zip)
  - removing in documentation references to ASR example, a.k.a. speech_recognition.

Closes https://xmosjira.atlassian.net/browse/AP-293

### CODE REVIEW

File: README.rst
Comment (mbanth) [inline README.rst:34]: This description says, `example designs ... example design`.  I suggest removing the 2nd phrase.

File: doc/programming_guide/ffd/speech_recognition_cyberon.rst
Comment (mbanth) [inline doc/programming_guide/ffd/speech_recognition_cyberon.rst:26]: It looks like the word `replace` was itself replaced with `sln_voice_asr_programming_guide` accidentally.

File: doc/programming_guide/ffd/speech_recognition_cyberon.rst
Comment (mbanth) [inline doc/programming_guide/ffd/speech_recognition_cyberon.rst:25]: Will this ToDo be completed as part of this PR?  If not, let's capture it in a Jira issue somewhere.

File: doc/programming_guide/ffd/speech_recognition_sensory.rst
Comment (mbanth) [inline doc/programming_guide/ffd/speech_recognition_sensory.rst:29]: Same commant as ToDo in previous file.

File: doc/programming_guide/low_power_ffd/speech_recognition.rst
Comment (mbanth) [inline doc/programming_guide/low_power_ffd/speech_recognition.rst:46]: And one more.

**Review by mbanth — CHANGES_REQUESTED** [review]

File: README.rst
Comment (ed-xmos) [inline README.rst:38]: I would call this two example designs bridging between TDM16 slave or USB Audio

File: doc/quick_start_guide/index.rst
Comment (ed-xmos) [inline doc/quick_start_guide/index.rst:21]: At least we covered the lack of quickstart in the v2.2.0

File: doc/programming_guide/ffd/speech_recognition_cyberon.rst
Comment (ed-xmos) [inline doc/programming_guide/ffd/speech_recognition_cyberon.rst:26]: I think you know this needs resolving before approval

File: tools/ci/build_examples.sh
Comment (ed-xmos) [inline tools/ci/build_examples.sh:80]: Good catch

**Review by mbanth — APPROVED** [review]

**Pull request merged**
